docs/src/markdown.ts

```diff
@@ -0,0 +1,266 @@
+import fs from "fs/promises";
+import path from "path";
+import { unified } from "unified";
+import remarkParse from "remark-parse";
+import remarkRehype from "remark-rehype";
+import rehypeRaw from "rehype-raw";
+import { visit } from "unist-util-visit";
+import GitHubSlugger from "github-slugger";
+import matter from "gray-matter";
+
+export interface Document {
+  /** the Markdown file itself, without from-matter */
+  content: string;
+
+  /** the path to this markdown file */
+  path: string;
+
+  /** the headings found in this markdown file */
+  headings: string[];
+
+  frontMatter: {
+    title: string;
+    description: string;
+  };
+}
+
+export type ErrorType = "link" | "hash" | "source" | "related";
+
+export type LinkError = {
+  type: ErrorType;
+  href: string;
+  doc: Document;
+};
+
+export const DOCS_PATH = ".";
+export const EXCLUDED_HASHES = ["top"];
+
+const slugger = new GitHubSlugger();
+
+/** Collect the paths of all .mdx files in the passed directories */
+const getAllMdxFilePaths = async (
```

Comment:
I'd love to get a test for this to make sure if follows symlinks

Reply:
update: this was overhauled in https://github.com/vercel/turborepo/commit/ef2d43be6bda7984cec9dc321e3e9d0342a1d715 and also now symlinks on mac work